app/controllers/api/cookies_consent_controller.rb

```diff
@@ -0,0 +1,26 @@
+module Api
+  class CookiesConsentController < BaseController
+    include ActionController::Cookies
+    respond_to :json
+
+    def show
+      render json: { cookies_consent: cookies_consent.exists? }
+    end
+
+    def create
```

Comment:
@luisramos0 Why do we request the server to set a cookie? Can't we just do it via Javascript? https://docs.angularjs.org/api/ngCookies/service/$cookies That could be a lot simpler and faster.

Reply:
An e-commerce website should only use httponly cookies that cannot be set by the browser. It's a simple security measure to avoid a million types of XSS attacks. That's why you should always go around to the server like here.
BUT in the end, I forgot to actually set it to httponly, which is as simple as adding a line like this to the cookie setting statement:
:secure => true

Maybe I should create a new issue to make this a secure cookie? What do you think?

Reply:
Wow, thank you for that insight. Even though it's not crucial for this cookie, we should definitely use HttpOnly cookies everywhere. Go ahead and create that issue.

Reply:
Issue #2554  with PR #2555 in code review.